release/models/platform/openconfig-platform.yang

```diff
@@ -407,6 +413,14 @@ module openconfig-platform {
         (field replaceable unit)";
     }
 
+    leaf clei-code {
+      type string;
+      description
+        "Common Language Equipment Identifier (CLEI) code of the
```

Comment:
Is there a reference for CLEI? It'd be good to know whether there is more than just a `string` format for it, but I wasn't sure where to look it up :-)

Reply:
It is covered as part of ANSI ATIS-0300213.2014 however these are not freely open accessible documents

https://www.commonlanguage.com/resources/commonlang/productshowroom/showroom/equip_id/carriers/eqpt_ja_501_i1.pdf

```
1. What is a Common Language Equipment Identification (CLEI) Code?
CLEI Codes are globally unique, 10-character alphanumeric intelligent codes that
identify equipment in a structured naming format. There is a one-to-one relationship
between a CLEI Code and a Manufacturer’s Product Code (part number including
the hardware version).
```

So it could likely be scoped if necessary but unsure if all implementations conform to this tightly

Reply:
Hi @earies - if the CLEI code has a relationship with the manufacturer's product code and there is already a "part-no"  leaf and a "hardware-version" leaf that can be used to identify the CLEI code in the provider's backend inventory/asset management system.  It is sort of like the CLLI code that is covered by the "location" leaf.

Reply:
This sounds sensible, but it seems it is non trivial to automatically derive the CLEI code from part-no leaf and hardware-version when the device could provide it directly.  I am not a subject matter expert on CLEI data, so my opinion is not very informed.

Reply:
On the config side that makes sense to me as if we wanted to configure a card type for instance, part-no and potentially hardware-version are all that would be needed. We wouldn't want to have an additional config leaf which could lead to unnecessary configuration permutations. I definitely think we should push back on having CLEI be used for config for that reason.

In this case, this is just a state leaf. Some operators do use CLEI codes and having to figure out this mapping on the operator side seems like work we can remove by letting the device's report this value. Please let me know if this makes sense.